lib/dgram.js

```diff
@@ -272,6 +267,41 @@ Socket.prototype.send = function(buffer,
   if (offset + length > buffer.length)
     throw new RangeError('Offset + length beyond buffer length');
 
+  return buffer.slice(offset, offset + length);
+}
+
+Socket.prototype.send = function(buffer,
+                                 offset,
+                                 length,
+                                 port,
+                                 address,
+                                 callback) {
+  var self = this;
+
+  if (arguments.length === 5 || arguments.length === 6) {
+    buffer = sliceBuffer(buffer, offset, length);
```

Comment:
I agree, that was done before and I cut it from here to get some feedback. However, I am struggling a bit to support both APIs through the same C++ function.
Should I (partially) duplicate that, or should I use positional argument in the array for offset and length for each buffer?
Or should I allocate a new object for each of this? You probably have some idea of what will behave better.

Basically I can pass `[buf1, offset1, length1, buf2, offset2, length2, ...]`. 

However I am not entirely sure this is worth the effort, but me and @mscdex disagree on this: https://github.com/nodejs/node/issues/4302#issuecomment-165295080 https://github.com/nodejs/node/issues/4302#issuecomment-165387075